packages/xforms-engine/src/parse/body/control/ItemsetDefinition.ts

```diff
@@ -49,7 +54,19 @@ export class ItemsetDefinition extends BodyElementDefinition<'itemset'> {
 			throw new Error(`<itemset> has no <value>`);
 		}
 
-		this.value = new ItemsetValueExpression(this, valueExpression);
 		this.label = ItemsetLabelDefinition.from(form, this);
+		this.value = new ItemsetValueExpression(this, valueExpression);
+		this.geometry = new ItemsetGeometryExpression(this);
+
+		this.metadataExclusions = ['itextId', this.value.expression, this.geometry.expression];
```

Comment:
As noted earlier, I aim to exclude anything already exposed as a class attribute to prevent data duplication and simplify the Vue client’s looping and rendering work.